app/models/user.rb

```diff
@@ -78,12 +78,13 @@ def four_different_characters
 
   has_many :team_users
   has_many :teams, through: :team_users, source: :team
-  has_many :non_single_teams, -> { where single: false }, through: :team_users, source: :team
+  has_many :teams_created_by_user, -> { where created_by_event: false }, through: :team_users, source: :team
+  has_many :teams_created_by_event, -> { where created_by_event: true }, through: :team_users, source: :team
```

Comment:
Is `:teams_created_by_event` used anywhere? If not, it could be removed for the moment imho.